src/attribute_type.rs

```diff
@@ -396,567 +307,593 @@ impl SdpAttributeSsrc {
+            message: "Unsupported sctpmap type token".to_string(),
+            line: line.to_string()})
+    }
+    Ok(SdpAttribute::Sctpmap(
+        SdpAttributeSctpmap {
+            port,
+            channels: tokens[2].parse::<u32>()?
+        }
+    ))
+}
+
+fn parse_setup(to_parse: &str,
+               line: &str) -> Result<SdpAttribute, SdpParserError>  {
+    Ok(SdpAttribute::Setup(
+        match to_parse.to_lowercase().as_ref() {
+            "active" => SdpAttributeSetup::Active,
+            "actpass" => SdpAttributeSetup::Actpass,
+            "holdconn" => SdpAttributeSetup::Holdconn,
+            "passive" => SdpAttributeSetup::Passive,
+            _ => return Err(SdpParserError::Line{
+                message: "Unsupported setup value".to_string(),
+                line: line.to_string()}),
         }
+    ))
+}
+
+fn parse_simulcast(to_parse: &str,
+                   line: &str) -> Result<SdpAttribute, SdpParserError>  {
+    let mut tokens = to_parse.split_whitespace();
+    let mut token = match tokens.next() {
+        None => return Err(SdpParserError::Line{
+            message: "Simulcast attribute is missing send/recv value".to_string(),
+            line: line.to_string()}),
+        Some(x) => x,
+    };
+    let mut sc = SdpAttributeSimulcast {
+        send: Vec::new(),
+        receive: Vec::new()
+    };
+    loop {
+        let sendrecv = match token.to_lowercase().as_ref() {
+            "send" => SdpAttributeDirection::Sendonly,
+            "recv" => SdpAttributeDirection::Recvonly,
+            _ => return Err(SdpParserError::Line{
+                message: "Unsupported send/recv value in simulcast attribute".to_string(),
+                line: line.to_string()}),
+        };
+        match tokens.next() {
+            None => return Err(SdpParserError::Line{
+                message: "Simulcast attribute is missing id list".to_string(),
+                line: line.to_string()}),
+            Some(x) => sc.parse_ids(sendrecv, x.to_string()),
+        };
+        token = match tokens.next() {
+            None => { break; },
+            Some(x) => x,
+        };
+    }
+    Ok(SdpAttribute::Simulcast(sc))
+}
+
+fn parse_ssrc(to_parse: &str,
+              line: &str) -> Result<SdpAttribute, SdpParserError>  {
+    let mut tokens  = to_parse.split_whitespace();
+    let ssrc_id = match tokens.next() {
+        None => return Err(SdpParserError::Line{
+            message: "Ssrc attribute is missing ssrc-id value".to_string(),
+            line: line.to_string()}),
+        Some(x) => x.parse::<u32>()?
     };
-    let mut attr = SdpAttribute::new(attrtype);
-    attr.parse_value(val.trim())?;
+    let mut ssrc = SdpAttributeSsrc::new(ssrc_id);
+    match tokens.next() {
+        None => (),
+        Some(x) => ssrc.set_attribute(x),
+    };
+    Ok(SdpAttribute::Ssrc(ssrc))
+}
+
+pub fn parse_attribute(value: &str) -> Result<SdpLine, SdpParserError> {
+    let attr: SdpAttribute = value.parse()?;
```

Comment:
I think we still want the `trim()` call to strip of white spaces. Could be either `value.trim().parse()` or you do the trimming inside of the `from_str()` function.

I'm also wondering if you could combine the two lines into a single line here.